From 8036ee340676f6a078d3371cf5f0ffcb56f896ed Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Sun, 21 Apr 2019 08:19:31 +0200
Subject: [PATCH 39/75] trace: annotate format_mod_{start,stop}

HANDLED:

kernel/trace/trace_printk.c:164:1: warning: mutex 'btrace_mutex' is still held at the end of function [-Wthread-safety-analysis]
}
^
kernel/trace/trace_printk.c:163:2: note: mutex acquired here
        mutex_lock(&btrace_mutex);
        ^
kernel/trace/trace_printk.c:168:2: warning: releasing mutex 'btrace_mutex' that was not held [-Wthread-safety-analysis]
        mutex_unlock(&btrace_mutex);
        ^

NOW OPEN:

kernel/trace/trace_printk.c:291:1: warning: mutex 'btrace_mutex' is still held at the end of function [-Wthread-safety-analysis]
}
^
kernel/trace/trace_printk.c:289:2: note: mutex acquired here
        format_mod_start();
        ^
kernel/trace/trace_printk.c:338:2: warning: releasing mutex 'btrace_mutex' that was not held [-Wthread-safety-analysis]
        format_mod_stop();
        ^
2 warnings generated.
---
 kernel/trace/trace_printk.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/trace/trace_printk.c b/kernel/trace/trace_printk.c
index c3fd849d4a8f..780fe634be12 100644
--- a/kernel/trace/trace_printk.c
+++ b/kernel/trace/trace_printk.c
@@ -158,12 +158,12 @@ find_next_mod_format(int start_index, void *v, const char **fmt, loff_t *pos)
 	return &mod_fmt->fmt;
 }
 
-static void format_mod_start(void)
+static void format_mod_start(void) __attribute__((acquire_capability(btrace_mutex)))
 {
 	mutex_lock(&btrace_mutex);
 }
 
-static void format_mod_stop(void)
+static void format_mod_stop(void) __attribute__((release_capability(btrace_mutex)))
 {
 	mutex_unlock(&btrace_mutex);
 }
-- 
2.17.1

